### ISSUE TITLE
Reprojucer.cmake doesn't support the setting USE_IPP_LIBRARY

### ISSUE THREAD

**Issue by McMartin** [McMartin/FRUT#252]
If you would like `Reprojucer.cmake` to support the setting `USE_IPP_LIBRARY` ("Use IPP Library" in Projucer), please leave a comment on this issue.

**Comment by vallant** [comment]
Hi, we would love to try out FRUT for our plugin, but we in fact need IPP! Is there a possibility to add this? :)
This is a great project, it always bothered me that it is necessary to use the Projucer instead of the de-facto standard build system CMake.

Also in perspective of unit testing and CI, the Projucer just leaves too little flexibility.

**Comment by McMartin** [comment]
Hi @vallant,
Thank you for considering FRUT!

IPP is a bit tricky to support since Projucer simply adds a project attribute (https://github.com/WeAreROLI/JUCE/blob/49c631e916a9f0c747348e6fcd092dc6283e0fe9/extras/Projucer/Source/ProjectSaving/jucer_ProjectExport_MSVC.h#L434) and lets Visual Studio do all the actual work :smile:

Could you please compare the compiler and linker command lines with "Use IPP Library" enabled and disabled, and report the difference in a comment here? That would be a great starting point for me. 
If you don't know how to do this, just tell me and I'll guide you through.

I could also install IPP on my computer and do that comparison myself, but I think it is better to involve users of FRUT into its development.

**Comment by vallant** [comment]
Hi, thanks for the quick reply!

I'll have a look soon, but now that i think of it, using IPP cannot be more 
than linking an additional library, so i guess by setting a IPP_HOME 
variable it should be possible to do that. I'll check it soon!

**Comment by vallant** [comment]
Ok this is what i found out- the only thing the `USE_IPP_LIBRARY` changes are the include and library directories, when enabling IPP is that `$(IPPIncludeDir);` is added to the include Directories and `$(IPPLibDir);` is added to the library directories. Note that the lib is only added to the shared code and inherited from the plugin-projects, while the include is added for all projects.

This also is in line what is stated on their homepage: 
https://software.intel.com/en-us/ipp-dev-guide-configuring-the-microsoft-visual-studio-ide-to-link-with-intel-ipp

On OSX this is done by default, there exists no `USE_IPP_LIBRARY`, as you just install IPP which is then added to the system include/library paths (from what i've seen)

**Comment by McMartin** [comment]
@vallant thanks for this very useful information! However, it seems that we still have 8 cells to fill in the following table:

| "Use IPP Library" setting in Projucer  | `<UseIntelIPP>...</UseIntelIPP>` setting in .vcxproj file | Include Directories | Library Directories |
|--------------------------------|----------------------|---------------------|-----------------|
| Yes (Default Linking)          | `true`               | `$(IPPIncludeDir);` | `$(IPPLibDir);` |
| Multi-Threaded Static Library  | `Parallel_Static`    |                     |                 |
| Single-Threaded Static Library | `Sequential`         |                     |                 |
| Multi-Threaded DLL             | `Parallel_Dynamic`   |                     |                 |
| Single-Threaded DLL            | `Sequential_Dynamic` |                     |                 |

Let me know if you would prefer me to take care of it.

**Comment by McMartin** [comment]
I finally installed Intel IPP on my computer, and did all the experiments I needed to understand what exactly Visual Studio is doing when changing the "Use Intel® IPP" setting. I found out that a compile definition (either `_IPP_SEQUENTIAL_DYNAMIC`, `_IPP_PARALLEL_DYNAMIC`, `_IPP_SEQUENTIAL_STATIC`, or `_IPP_PARALLEL_STATIC`) is passed to the compiler, in order to then link against the right version of the IPP libraries.

I combined these findings with figuring out how to get the Intel IPP installation directory from the Windows registry, and I opened a PR: https://github.com/McMartin/FRUT/pull/505. Feel free to review it!
